api/doc/thor.yaml

```diff
@@ -2259,6 +2327,13 @@ components:
       schema:
         type: string
 
+    NewestBlockInQuery:
```

Comment:
This seems to be the same concept as `revision`, wdyt?

Reply:
Coming back to this since I found that the API in ethereum only support block number and tags but node block ID. I would think it's not necessary to work with block ID in this API.

Reply:
you mean a type that is essentially `revision` without blockID? so:

```
func ParseRevision(revision string, allowNext bool) (*Revision, error) {
	if revision == "" || revision == "best" {
		return &Revision{revBest}, nil
	}

...
        // WITHOUT THIS
	if len(revision) == 66 || len(revision) == 64 {
		blockID, err := thor.ParseBytes32(revision)
		if err != nil {
			return nil, err
		}
		return &Revision{blockID}, nil
	}

...
	return &Revision{uint32(n)}, err
}
```

Reply:
FYI `newestBlock` is indeed a `revision`, the name change is:

1. Because having `blockCount` and `revision` is confusing (going upwards or downwards?)
2. Because `newestBlock` is the name in Ethereum

Reply:
> you mean a type that is essentially revision without blockID? 
Yes, just think of the necessity of having a blockID as the argument.

Reply:
Ok, since the name of the new parser might be opinionated, and `ParseNewestBlock` does not seem generic enough, do you have any suggestions? The logic should be reused by the current `ParseRevision` IMO since they share common code.

To sum up, any suggestions abut the method name to parse this?

Reply:
Called it for now `ParseRevisionWithoutBlockID`, thanks